src/main/java/com/fasterxml/jackson/core/io/SegmentedStringWriter.java

```diff
@@ -82,7 +89,7 @@ public Writer append(CharSequence csq, int start, int end) {
      * @return String that contains all aggregated content
      */
     public String getAndClear() {
-        String result = _buffer.contentsAsString();
+        String result = _buffer.contentsAsString(unlimitedConstraints);
```

Comment:
As per my other comment I don't think we should guard this particular entry point; hence no need to pass no-limits constraints. (meaning, nothing would be passed or checked at this point)